src/prefect/orion/database/interface.py

```diff
@@ -79,11 +80,34 @@ async def engine(self):
 
     async def session(self):
         """
-        Provides a SQLAlchemy session
+        Provides a SQLAlchemy session.
         """
         engine = await self.engine()
         return await self.database_config.session(engine)
 
+    @asynccontextmanager
+    async def session_context(self):
```

Comment:
To resolve the two-context-manager confusion, perhaps
```suggestion
    async def session_context(self, begin_transaction:bool=False):
```

Reply:
I kind of like that. Then it can just be `with db.session() as session:` or `with db.session(begin_transaction=True)` or `with_transaction=True`.

Reply:
I like this, I've implemented it

Reply:
Is there a case where you should use `session` without a context manager or should we just have `session` be a context and have something else for the lower-level session retrieval?